### DISCUSSION TITLE
Add leading zero capabilities to number display function

### DISCUSSION THREAD

**Discussion by erdosip** [Bodmer/TFT_eSPI#1848] [New processor/display support or capability request] [answered]
Hy!

I would like an improvement to the function drawNumber() (and maybe for drawFloat() )
My problem is that I used the LCDWIKI library, but now I need to change my code to the TFT_eSPI library. The porting goes well, but I faced an issue with numbers. With the LCDWIKI library, I had  a function to print integers, which can get a "length" and a "filler" parameter along with the coordinates, and the integer itself.

The length is a minimum number of digits/chars to display
The filler is the char which is used to fill up the digit to the length.

So for example, is I wanted to have my month of the year displayed in two digit format, i did this:
```cpp
  DummyLCD.Print_Number_Int(now.month(), 84, 287, 3, '0', 10);

```
The 84/287 are the coordinates, the 3 is the lenght (which include the not shown sign, if I'm right) and the '0' is the leading char (but you can use space, or any other char), which is included before the actual number, to get the proper constant lenght, and let the text displayed staying at the same place.

I know, that I can code this in my own code, but if the library can support it, that would be more elegant I think.

The same can be applied to drawFloat too, where the decimals are good, but if I want to display 1.234566 or 11.234567 the number will be moving.

Thanks,
 Peter

**Comment by Bodmer** [comment, marked as answer]
The inclusion of drawNumber and drawFloat is historical for board packages that did not include printf() or sprintf(). You can use these for example:
```
  tft.setCursor(x, y);
  tft.printf("%02d:%02d", 3, 4);

  char str[10];
  sprintf(str, "%02d:%02d", 3, 4);
  tft.drawString(str, x, y);
```
Since the printf function is so flexible it does not seem worth adding another function variant to the library.

For floating point numbers it is best to draw the value with right justification, then the decimal point stays in the same place (caveat is that the digits are monospaced), i.e. use the setTextDatum() function plus setTextPadding() to wipe leading characters as required.

**Reply by erdosip** [reply]
It's works like a charm! Thanks to pointing this out!
